df_auth/backends.py

```diff
@@ -9,163 +14,204 @@
 User = get_user_model()
 
 
-class EmailOTPBackend(ModelBackend):
-    @staticmethod
-    def get_users(email):
-        return User.objects.filter(is_active=True, email=email)
-
-    def authenticate(self, request=None, email=None, otp=None, **kwargs):
-        if email and otp:
-            for user in self.get_users(email):
-                if user.email == api_settings.TEST_USER_EMAIL:
-                    return user
-                if self.user_can_authenticate(user):
-                    devices = EmailDevice.objects.filter(user=user, email=email)
-                    return self.authenticate_devices(devices, user, otp)
-
-    def generate_challenge(
-            self, request=None, user=None, email=None, extra_context=None, **kwargs
-    ):
-        if not email:
-            return None
-
-        if request.user.is_authenticated:
-            device = EmailDevice.objects.filter(user=request.user, email=email).first()
-            if not device and self.get_users(email):
-                raise ValidationError(
-                    _("User with this email is already exist")
-                )
-            users = [request.user]
-        else:
-            users = [user] if user else self.get_users(email)
+def ensure_backend_effective(method):
+    @functools.wraps(method)
+    def wrapper(self, *args, **kwargs):
+        if self.is_backend_effective(**kwargs):
+            return method(self, *args, **kwargs)
+
+    return wrapper
+
+
+class TestEmailBackend(ModelBackend):
+    def authenticate(self, request, **kwargs):
+        if api_settings.TEST_USER_EMAIL and kwargs.get("email") == api_settings.TEST_USER_EMAIL:
+            return User._default_manager.get(email=api_settings.TEST_USER_EMAIL)
+
+
+class BaseOTPBackend(ModelBackend):
+    identity_field: str
+    device_identity_field: str
+    DeviceModel: Type[SideChannelDevice]
+
+    DEVICE_TAKEN_MESSAGE = "This device is already taken"
```

Comment:
These should be codes, not messages, i.e.

device_already_exists
user_already_exists

etc